extensions/logs/suite.go

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logs
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+var config Config
+
+type Config struct {
+	ArtifactsDir   string        `default:"logs" desc:"Directory for storing container logs" split_words:"true"`
+	ExcludeK8sNs   string        `default:"kube-system,local-path-storage" desc:"Comma-separated list of excluded kubernetes namespaces" split_words:"true"`
+	ContextTimeout time.Duration `default:"15s" desc:"Context timeout for kubernetes queries" split_words:"true"`
+	LogWorkerCount int           `default:"4" desc:"Number of log collector workers" split_words:"true"`
+}
+
+type Suite struct {
+	suite.Suite
+
+	kubeClient kubernetes.Interface
+
+	testStartTime time.Time
+	nsmContainers map[types.UID]bool
+	logQueue      chan logItem
+	waitGroup     sync.WaitGroup
+}
+
+type logItem struct {
+	namespace  string
+	pod        string
+	logDir     string
+	logOptions *corev1.PodLogOptions
+}
+
+func init() {
+	if err := envconfig.Usage("", &config); err != nil {
+		panic(err)
+	}
+
+	if err := envconfig.Process("", &config); err != nil {
+		panic(err)
+	}
+}
```

Comment:
I think we can use `once` and do lazy initialization. Also, we can do there setup for the worker pool.